plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DesktopApplicationImpl.java

```diff
@@ -17,14 +17,97 @@
 package org.jkiss.dbeaver.registry;
 
 import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.equinox.app.IApplicationContext;
+import org.eclipse.equinox.internal.security.auth.AuthPlugin;
+import org.eclipse.equinox.internal.security.storage.StorageUtils;
+import org.eclipse.osgi.service.environment.EnvironmentInfo;
 import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.Log;
 import org.jkiss.dbeaver.model.app.DBPPlatform;
 import org.jkiss.dbeaver.model.app.DBPWorkspaceDesktop;
+import org.jkiss.dbeaver.utils.RuntimeUtils;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.Arrays;
 
 /**
  * EclipseApplicationImpl
  */
 public abstract class DesktopApplicationImpl extends BaseApplicationImpl {
+    private static final Log log = Log.getLog(DesktopApplicationImpl.class);
+
+    private static final String DBEAVER_DATA_DIR = "DBeaverData"; //$NON-NLS-1$
+    private static final String DBEAVER_SECURE_DIR = "secure"; //$NON-NLS-1$
+    private static final String DBEAVER_SECURE_FILE = "secure_storage"; //$NON-NLS-1$
+    private static final String ECLIPSE_KEYRING = "-eclipse.keyring"; //$NON-NLS-1$
```

Comment:
You can get this constant from org.eclipse.equinox.internal.security.storage.SecurePreferencesMapper

Reply:
It has private access there.